pkg/rtc/participant.go

```diff
@@ -434,7 +451,41 @@ func (p *ParticipantImpl) Close() error {
 }
 
 func (p *ParticipantImpl) Negotiate() {
-	p.subscriber.Negotiate()
+	if p.MigrateState() != types.MigrateStateInit {
+		p.subscriber.Negotiate()
+	}
+}
+
+func (p *ParticipantImpl) InitSubscribePreviousAnwser(previous *webrtc.SessionDescription) {
+	p.subscriber.SetInitPreviousAnwser(previous)
+}
+
+func (p *ParticipantImpl) SetMigrateState(s types.MigrateState) {
+	p.lock.Lock()
+	preState := p.MigrateState()
+	if preState == types.MigrateComplete || preState == s {
+		p.lock.Unlock()
+		return
+	}
+	p.params.Logger.Debugw("SetMigrateState", "state", s)
+	p.migrateState.Store(s)
+	if s == types.MigrateStateSync {
+		if !p.UptrackManager.HasPendingMigratedTrack() {
+			p.migrateState.Store(types.MigrateComplete)
+		}
+		pendingOffer := p.pendingOffer
+		p.pendingOffer = nil
+		p.lock.Unlock()
```

Comment:
```suggestion
     }
    p.lock.Unlock()
    if pendingOffer != nil {
        p.HandleOffer(*pendingOffer)
    }
```
Just to prevent an else with just an unlock.